File: backend/src/main/java/net/archasmiel/skufapi/api/request/auth/GoogleAuthRequest.java
package net.archasmiel.skufapi.api.request.auth;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.archasmiel.skufapi.api.request.ApiRequest;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Sign in request via Google")
public class GoogleAuthRequest implements ApiRequest {

    @Schema(description = "GoogleId token", example = "eyJhbGciOiJIUzUxMiJ9...")
    private String token;
}

File: backend/src/main/kotlin/net/archasmiel/skufapi/api/request/auth/GoogleAuthRequest.kt
package net.archasmiel.skufapi.api.request.auth

import io.swagger.v3.oas.annotations.media.Schema
import net.archasmiel.skufapi.api.request.ApiRequest

@Schema(description = "Login request via Google")
data class GoogleAuthRequest(
    @field:Schema(description = "Google id token", example = "eyJhbGciOiJIUzUxMiJ9...")
    val token: String
) : ApiRequest